File: backend/scripts/seed_from_json.py
"""Create tables and seed PostgreSQL from platform/data/cms.json."""

import json
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(ROOT))

from app.database import Base, SessionLocal, engine  # noqa: E402
from app.schemas.cms import CmsDataSchema  # noqa: E402
from app.services.cms import load_cms, save_cms  # noqa: E402

CMS_JSON = ROOT.parent / "platform" / "data" / "cms.json"


def main() -> None:
    print("Creating tables...")
    Base.metadata.create_all(bind=engine)

    if not CMS_JSON.exists():
        print(f"Missing {CMS_JSON}")
        sys.exit(1)

    raw = json.loads(CMS_JSON.read_text(encoding="utf-8"))
    payload = CmsDataSchema.model_validate(raw)

    db = SessionLocal()
    try:
        print("Seeding CMS data...")
        save_cms(db, payload)
        result = load_cms(db)
        print(
            f"Done: {len(result.states)} states, {len(result.cities)} cities, "
            f"{len(result.lawyers)} lawyers"
        )
    finally:
        db.close()


if __name__ == "__main__":
    main()

File: backend-node/scripts/seed-from-json.ts
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { pool } from '../src/db.js';
import { loadCms, saveCms } from '../src/cms.js';
import { DEFAULT_SUBSCRIPTION_PLANS } from '../src/subscription-plans.js';
import type { CmsData } from '../src/types.js';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const cmsJson = path.join(__dirname, '..', '..', 'platform', 'data', 'cms.json');
const lawyersSample = path.join(__dirname, '..', '..', 'platform', 'data', 'lawyers-sample.json');

async function main() {
  if (!fs.existsSync(cmsJson)) {
    console.error(`Missing ${cmsJson}`);
    process.exit(1);
  }

  const raw = JSON.parse(fs.readFileSync(cmsJson, 'utf-8')) as CmsData;
  if (fs.existsSync(lawyersSample)) {
    raw.lawyers = JSON.parse(fs.readFileSync(lawyersSample, 'utf-8')) as CmsData['lawyers'];
    console.log(`Loaded ${raw.lawyers.length} lawyers from lawyers-sample.json`);
  }
  if (!raw.subscriptionPlans?.length) {
    raw.subscriptionPlans = DEFAULT_SUBSCRIPTION_PLANS;
  }
  if (!raw.siteContent) {
    const { defaultSiteContent } = await import('../src/default-site-content.js');
    raw.siteContent = defaultSiteContent as CmsData['siteContent'];
  }
  console.log('Seeding CMS data from cms.json...');
  await saveCms(raw);
  const result = await loadCms();
  console.log(
    `Done: ${result.states.length} states, ${result.cities.length} cities, ${result.lawyers.length} lawyers`,
  );
  await pool.end();
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
